inc/Integrations/Airtable/AirtableIntegration.php

```diff
@@ -2,34 +2,45 @@
 
 namespace RemoteDataBlocks\Integrations\Airtable;
 
-use RemoteDataBlocks\Logging\LoggerManager;
 use RemoteDataBlocks\WpdbStorage\DataSourceCrud;
 
 class AirtableIntegration {
 	public static function init(): void {
-		$data_sources = DataSourceCrud::get_data_sources( REMOTE_DATA_BLOCKS_AIRTABLE_SERVICE );
+		$data_source_configs = DataSourceCrud::get_configs_by_service( REMOTE_DATA_BLOCKS_AIRTABLE_SERVICE );
 
-		foreach ( $data_sources as $config ) {
-			self::register_blocks_for_airtable_data_source( $config );
+		foreach ( $data_source_configs as $config ) {
+			$data_source = AirtableDataSource::from_array( $config );
+			self::register_block_for_airtable_data_source( $data_source );
 		}
 	}
 
-	private static function register_blocks_for_airtable_data_source( array $config ): void {
-		/** @var AirtableDataSource $airtable_data_source */
-		$airtable_data_source = AirtableDataSource::from_array( $config );
-
-		$block_name = $airtable_data_source->get_display_name();
-		$query = $airtable_data_source->___temp_get_query();
-		$list_query = $airtable_data_source->___temp_get_list_query();
-
-		if ( is_wp_error( $query ) || is_wp_error( $list_query ) ) {
-			LoggerManager::instance()->error( 'Failed to get query for Airtable block' );
-			return;
-		}
+	public static function register_block_for_airtable_data_source( AirtableDataSource $data_source, array $block_overrides = [] ): void {
```

Comment:
Should we follow the `{type}_block` naming here and make this something like `register_item_block_...` or `register_single_item_block_...` ?